### PR TITLE
[Sidebar: 3 of 4] Implement docking sidebar

### PR DESCRIPTION
**Pull request by jfurrow**
~~This includes changes from #1412 and #1417.~~ These have merged :)

The sidebar can now be docked and undocked from the UI at viewports above our medium breakpoint. When docked (default state), it behaves the same as the current UI. When undocked, the sidebar is hidden off screen and its visibility can be toggled by tapping the menu icon in the page header. The docked state can be toggled by clicking the icon at the bottom of the sidebar or pushing the left bracket key (`[`).

### CODE REVIEW

**Comment by jfurrow** [comment]
retest this please

**Comment by orlandohohmeier** [comment]
@jfurrow please rebase ... ♻️ ⚾️

**Comment by jfurrow** [comment]
@orlandohohmeier @MatApple Finally got this rebased! I also noticed a potential issue with the positioning of the button so I've made adjustments accordingly. Should be good to go now :)

**Comment by orlandohohmeier** [comment]
I'm not to sure about the sidebar toggle button... Shouldn't it stay so that one can use it to dock the sidebar if undocked?

This comment aside, works greats and looks good. ✨ 💯

**Comment by jfurrow** [comment]
@orlandohohmeier The intention is for the toggle button in the page header to act as a visibility toggle for the sidebar. The button to toggle the docking of the sidebar is at the bottom right-hand of the sidebar.

File: src/js/stores/SidebarStore.js [hunk 1]
```diff
@@ -44,14 +46,26 @@ class SidebarStore extends GetSetBaseStore {
...
+              isDocked: nextDockedState
+            });
+            this.emitChange(SIDEBAR_CHANGE);
+          }
+          break;
         case REQUEST_SIDEBAR_CLOSE:
         case REQUEST_SIDEBAR_OPEN:
-          var oldIsOpen = this.get('isOpen');
-          var isOpen = action.data;
+          var oldisVisible = this.get('isVisible');
```
Comment (Poltergeist) [inline src/js/stores/SidebarStore.js]: #Nit of the day. `oldIsVisible` camel case.

**Pull request merged**